File: src/main/java/jp/ngt/rtm/modelpack/state/DataType.java
package jp.ngt.rtm.modelpack.state;

import java.util.Arrays;

public enum DataType {
	BOOLEAN("Boolean"),
	DOUBLE("Double"),
	INT("Int"),
	STRING("String"),
	VEC("Vec"),
	HEX("Hex"),
	;

	public final String key;

	DataType(String par1) {
		this.key = par1;
	}

	public static DataType getType(String s) {
		return Arrays.stream(DataType.values()).filter(type -> type.key.equals(s)).findFirst().orElse(null);
	}

	/*public interface DataEntryConverter
	{
		DataEntry toEntry(String type, String data, int flag);
	}*/
}

File: src/main/java/jp/ngt/rtm/modelpack/state/DataType.kt
package jp.ngt.rtm.modelpack.state

enum class DataType(@JvmField val key: String) {
    BOOLEAN("Boolean"),
    DOUBLE("Double"),
    INT("Int"),
    STRING("String"),
    VEC("Vec"),
    HEX("Hex"),
    LIST("List");

    companion object {
        @JvmStatic
        fun getType(s: String?) = entries.firstOrNull { type: DataType? -> type!!.key == s }
    }
}
